apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -129,12 +131,36 @@ where
 
         cache_keys.shuffle(&mut thread_rng());
 
+        let persisted_queries_operations = persisted_query_layer.all_operations();
+
+        tracing::info!(
+            "warming up the query plan cache with {} queries, this might take a while",
+            cache_keys.len()
+                + persisted_queries_operations
+                    .as_ref()
+                    .map(|ops| ops.len())
+                    .unwrap_or(0)
+        );
+
+        let mut all_cache_keys = Vec::new();
```

Comment:
can we initialize it with a capacity ? Like persisted_queries_operations.len() + cache_keys.len() ?

Reply:
done in [5f3df65](https://github.com/apollographql/router/pull/3829/commits/5f3df656486ddf2d86fd641a955029c8e75f4865)